pkg/credentialprovider/gcp/metadata.go

```diff
@@ -98,10 +106,20 @@ func init() {
 		})
 }
 
+// Returns true if it finds a local GCE VM.
+// Looks at a product file that is an undocumented API.
```

Comment:
Is it undocumented but expected to remain stable? Or undocumented as in it may change or break at any point?